solarforecastarbiter/io/reference_observations/tests/test_ref_eia.py

```diff
@@ -0,0 +1,178 @@
+    status = mocker.patch(
+        'solarforecastarbiter.io.api.APISession.create_observation')
+    eia.initialize_site_observations(session, site)
+    assert status.called
+
+
+@pytest.fixture
+def mock_obs_creation(mocker):
+    mocked = mocker.patch(
+        'solarforecastarbiter.io.reference_observations.common.'
+        'create_observation')
+    return mocked
+
+
+@pytest.fixture
+def mock_obs_creation_error(mocker, mock_obs_creation):
+    mock_obs_creation.side_effect = HTTPError(
+        response=mocker.Mock(response=mocker.Mock(text="eror")))
+
+
+def test_initialize_site_obs(mock_api, mock_obs_creation, site):
+    eia.initialize_site_observations(mock_api, site)
+    mock_obs_creation.assert_called()
+
+
+def test_initialize_site_obs_http_error(
+        log, mock_api, mock_obs_creation_error, site):
+    eia.initialize_site_observations(mock_api, site)
+    assert log.error.call_count == 1
+    assert log.debug.call_count == 1
+
+
+def test_fetch(mocker, session, site):
+    status = mocker.patch(
+        'solarforecastarbiter.io.fetch.eia.get_eia_data'
+    )
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+
+    index = pd.date_range(start, end, freq="1h")
+    data = {"net_load": range(len(index))}
+    df = pd.DataFrame(index=index, data=data)
+    status.return_value = df
+
+    out = eia.fetch(session, site, start, end, eia_api_key=api_key)
+    assert status.called
+    assert not out.empty
+
+
+def test_fetch_empty(log, mocker, session, site):
+    status = mocker.patch(
+        'solarforecastarbiter.io.fetch.eia.get_eia_data'
+    )
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+    status.return_value = pd.DataFrame()
+    out = eia.fetch(session, site, start, end, eia_api_key=api_key)
+    assert status.called
+    assert log.warning.call_count == 1
+    assert out.empty
+
+
+def test_fetch_fail(mocker, session, site_no_extra):
+    start = pd.Timestamp('2020-01-01T0000Z')
+    end = pd.Timestamp('2020-01-02T0000Z')
+    api_key = 'bananasmoothie'
+    out = eia.fetch(session, site_no_extra, start, end, eia_api_key=api_key)
+    assert out.empty
+
+
+def test_initialize_site_forecasts(
+        requests_mock, mocker, session, site, mock_list_sites):
+
+    obs = Observation(
+        site=site,
+        name="Net Load",
+        variable="net_load",
+        interval_length=pd.Timedelta("60min"),
+        interval_value_type="interval_mean",
+        interval_label="ending",
+        uncertainty=0,
+    )
+
+    mocker.patch('solarforecastarbiter.io.api.APISession.list_observations',
+                 return_values=[obs])
+    status = mocker.patch(
+        'solarforecastarbiter.io.api.APISession.create_forecast')
+    eia.initialize_site_forecasts(session, site)
+    assert status.called
```

Comment:
@alorenzo175 Adding a patch to the `APISession.list_observations` seems necessary as otherwise I get an error: `requests_mock.exceptions.NoMockAddress: No mock address: GET https://api.solarforecastarbiter.org/observations/`. With that patch, then the code runs but the `assert status.called` fails as it shows as False. Not entirely sure why, but also not sure if this overall approach is actually the "best" way to confirm that `eia.initialize_site_forecasts()` is working properly.

Reply:
ah that's from creating the persistence forecasts. Yes, I think this the best way to confirm initialize is working properly.

Reply:
as for the failing test, I think that will require stepping through ``ei.initialize_site_forecasts`` to see why it doesn't make it to create forecast

Reply:
Aaaggghhh... Turned out I had a typo in my `mocker.patch()` (`return_values=` instead of `return_value`), which then meant that nothing was correctly returned by the mocker.

Reply:
Figured out the necessary revisions to get the test to be valid. There's probably a more elegant way to do the patching than the `mocker.patch()` lines I've added, but it works.